RecoLocalCalo/HGCalRecProducers/src/ComputeClusterTime.cc

```diff
@@ -0,0 +1,204 @@
+  int totSize = t.size();
+
+  for (int ij = 0; ij <= start_el; ++ij) {
+    if (t[ij] > (t[start_el] - HalfTimeDiff)) {
+      for (int kl = ij; kl < totSize; ++kl) {
+        if (t[kl] < (t[end_el] + HalfTimeDiff)) {
+          sum += t[kl] * w[kl];
+          num += w[kl];
+        } else
+          break;
+      }
+      break;
+    }
+  }
+
+  if (num == 0) {
+    return std::pair<float, float>(-99., -1.);
+  }
+  return std::pair<float, float>(sum / num, 1. / sqrt(num));
+}
+
+/*
+std::pair<float,float> ComputeClusterTime::fixSizeHighestDensityEnergyResWeig(std::vector<float>& t,
+									      std::vector<float>& w,
+									      std::string& type,
+									      float deltaT = 0.210, //time window in ns
+									      float timeWidthBy = 0.5){
+  
+
+    //range is in SoN units
+    TF1* func = new TF1("func", timeResolution, 2., 1000., 2);
+    if(type == "test"){
+      //time is in ns units
+      func->SetParameters(5., 0.02);
+    }
+
+    std::vector<float> weights;
+    weights.resize(t.size());
+
+    for (unsigned int ij = 0; ij < w.size(); ++ij) {
+      float energy = w[ij];
+
+      if(energy > func->GetXmax()) weights[ij] = 1./func->GetParameter(1);
+      else if(energy < func->GetXmin()) weights[ij] = 1./func->GetParameter(0);
+      else weights[ij] = 1./func->Eval(energy);
+    }
+
+    return fixSizeHighestDensity(t, weights, deltaT, timeWidthBy);
+
+  }
+*/
+
+//useful for future developments - baseline for 0PU
+/*
+  //time-interval based on the smallest one containing a minimum fraction of hits
+  // vector with time values of the hit; fraction between 0 and 1; how much furher enlarge the selected time window
+  float highestDensityFraction(std::vector<float>& hitTimes, float fractionToKeep=0.68, float timeWidthBy=0.5){
+
+    std::sort(hitTimes.begin(), hitTimes.end());
+    int totSize = hitTimes.size();
+    int num = 0.;
+    float sum = 0.;
+    float minTimeDiff = 999.;
+    int startTimeBin = 0;
+
+    int totToKeep = int(totSize*fractionToKeep);
+    int maxStart = totSize - totToKeep;
+
+    for(int ij=0; ij<maxStart; ++ij){
+      float localDiff = fabs(hitTimes[ij] - hitTimes[int(ij+totToKeep)]);
+      if(localDiff < minTimeDiff){
+	minTimeDiff = localDiff;
+	startTimeBin = ij;
+      }
+    }
+
+    // further adjust time width around the chosen one based on the hits density
+    // proved to improve the resolution: get as many hits as possible provided they are close in time
+
+    int startBin = startTimeBin;
+    int endBin = startBin+totToKeep;
+    float HalfTimeDiff = std::abs(hitTimes[startBin] - hitTimes[endBin]) * timeWidthBy;
+
+    for(int ij=0; ij<startBin; ++ij){
+      if(hitTimes[ij] > (hitTimes[startBin] - HalfTimeDiff) ){
+	for(int kl=ij; kl<totSize; ++kl){
+	  if(hitTimes[kl] < (hitTimes[endBin] + HalfTimeDiff) ){
+	    sum += hitTimes[kl];
+	    ++num;
+	  }
+	  else  break;
+	}
+	break;
+      }
+    }
+
+    if(num == 0) return -99.;
+    return sum/num;
+  }
+  */
```

Comment:
I bet someone will, sooner or later, ask to remove commented-out code.

Reply:
OK, I also clean all this. 

Some of the commented-out code was kept in the first integration as useful for future developments. But it is superseded now by the rest of the code that is more general.
So I clean everything